src/Result.hpp

```diff
@@ -22,6 +22,7 @@
 
 #include "third_party/nonstd/optional.hpp"
 
+#include <algorithm>
```

Comment:
Suggestion: Move this to the `.cpp` file instead since it's not required by anything in the header file.